community/c++-algo/include/GameLib/src/gameMap.cpp

```diff
@@ -93,25 +94,115 @@ namespace terminal {
+                throw GameMapException("Error placing remove; 1 stationary unit not found");
+            // Clearing the vector for firewalls is not necessary, as we verified above
+            map.at(pos.x).at(pos.y).push_back(newUnit);
         }
+    }
+
+    /// Remove all GameUnits from the game map at the location. Throw an error if it's empty.
+    /// @param pos Position to clear from the game map
+    void GameMap::removeUnit(Pos pos) {
+        if (!inArenaBounds(pos)) throw PosException("Out of bounds exception");
+        if (map.at(pos.x).at(pos.y).size() == 0) throw GameMapException("Error trying to remove 0 units");
+        map.at(pos.x).at(pos.y).clear();
+    }
+
+    /// Takes a position and radius and fills a vector with locations in its range
+    /// @param locations A vector passed by reference to return the values.
+    /// @param pos The position to find locatins in range from.
+    /// @param radius The radius of the circle to get locations from.
+    void GameMap::getLocationsInRange(vector<Pos>& locations, Pos pos, double radius) {
+        if (radius < 0 or radius > ARENA_SIZE)
+            throw GameMapException("Error getting locations in range with that radius");
+        if (!inArenaBounds(pos)) throw PosException("Out of bounds exception");
+        for (int i = (int)(pos.x - radius); i < (int)(pos.x + radius + 1); i++) {
+            for (int j = (int)(pos.y - radius); j < (int)(pos.y + radius + 1); j++) {
+                Pos new_pos = { i, j };
+                if (inArenaBounds(new_pos) && distanceBetweenLocations(pos, new_pos) < radius + 0.51) {
+                    locations.push_back(new_pos);
+                }
+            }
+        }
+    }
+
+    /// Helper function to get the euclidean distance between two locations
+    /// @param pos_1 First position.
+    /// @param pos_2 Second position.
+    /// @return Euclidean distance between two positions.
+    double distanceBetweenLocations(Pos pos_1, Pos pos_2) {
+        return sqrt(pow(pos_1.x - pos_2.x, 2) + pow(pos_1.y - pos_2.y, 2));
+    }
+
+    /// Overloaded [Pos] operator enables access in the game map with a Pos.
+    /// @param pos Position to index into the map.
+    /// @return A reference to the vector of GameUnits at the location.
+    vector<GameUnit>& GameMap::operator[](const Pos &pos) {
+        return map.at(pos.x).at(pos.y);
+    }
+
+    /// Overloaded const [Pos] operator enables const access in the game map with a Pos.
+    /// @param pos Position to index into the map.
+    /// @return A reference to the vector of GameUnits at the location.
+    const vector<GameUnit>& GameMap::operator[](const Pos &pos) const {
+        return map.at(pos.x).at(pos.y);
+    }
+
+    /// Overloaded [int] operator enables access in the game map with [x][y]
+    /// @param x X coordinate to index into the map.
+    /// @return A reference to the column of the game map at coordinate X.
+    vector<vector<GameUnit> >& GameMap::operator[](int x) {

```

Comment:
This allows the user to index index with [x][y] in case that's their jam; just makes it a bit simpler to let them have more options.
```
for (int x = 0; .... 28)
  for (int y = 0; .... 28)
    game_map[x][y] // now valid
```

Reply:
I just that's true, I was just thinking in terms of only accessing the column, but you're right, this is a good addition.